pkg/gofr/datasource/pubsub/interface.go

```diff
@@ -19,7 +19,7 @@ type Subscriber interface {
 type Client interface {
 	Publisher
 	Subscriber
-	Health() datasource.Health
+	Health() health.Health
```

Comment:
when i brought in the go.mod into the nats package it created a import cycle I couldn't resolve w/o moving the health stuff into its own package.

Reply:
And by changing it we are getting these errors:
<img width="870" alt="Screenshot 2024-09-20 at 1 04 28 PM" src="https://github.com/user-attachments/assets/4375c127-e779-4a6e-bf74-1708b2e2a071">

Reply:
The health functionality has been used by other datasources also (ex: `cassandra`, `dgraph` etc) but none of them felt the need to export heath to a different package. This will create unnecessary complications. You only need a method that can check the Health of the data source so that we can call it here:

```go
func (c *Container) Health(ctx context.Context) interface{} {
	var (
		healthMap = make(map[string]interface{})
		downCount int
	)

	const statusDown = "DOWN"

	if !isNil(c.SQL) {
		health := c.SQL.HealthCheck()
		if health.Status == statusDown {
			downCount++
		}

		healthMap["sql"] = health
	}

	if !isNil(c.Redis) {
		health := c.Redis.HealthCheck()
		if health.Status == statusDown {
			downCount++
		}

		healthMap["redis"] = health
	}

	if c.PubSub != nil {
		health := c.PubSub.Health()
		if health.Status == statusDown {
			downCount++
		}

		healthMap["pubsub"] = health
	}

	downCount += checkExternalDBHealth(ctx, c, healthMap)

	for name, svc := range c.Services {
		health := svc.HealthCheck(ctx)
		if health.Status == statusDown {
			downCount++
		}

		healthMap[name] = health
	}

	c.appHealth(healthMap, downCount)

	return healthMap
}
```

Feel free to take a reference from them. Also when checking out your branch on local to test the code i am getting unwanted go.mod errors regarding this health package:

```
gofr.dev/pkg/gofr/datasource/pubsub/nats imports
	gofr.dev/pkg/gofr/health: module gofr.dev@latest found (v1.21.0), but does not contain package gofr.dev/pkg/gofr/health
```


please try to avoid this